src/Sylius/Component/Core/spec/Promotion/Applicator/OrderItemPromotionAdjustmentsApplicatorSpec.php

```diff
@@ -0,0 +1,130 @@
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Core\Promotion\Applicator\OrderItemPromotionAdjustmentsApplicator;
+use Sylius\Component\Core\Promotion\Applicator\OrderItemPromotionAdjustmentsApplicatorInterface;
+use Sylius\Component\Order\Factory\AdjustmentFactoryInterface;
+use Sylius\Component\Promotion\Model\PromotionInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ * @author Gorka Laucirica <gorka.lauzirika@gmail.com>
+ */
+final class OrderItemPromotionAdjustmentsApplicatorSpec extends ObjectBehavior
+{
+    function let(
+        AdjustmentFactoryInterface $adjustmentFactory
+    ) {
+        $this->beConstructedWith($adjustmentFactory);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(OrderItemPromotionAdjustmentsApplicator::class);
+    }
+
+    function it_implements_an_units_promotion_adjustments_applicator_interface()
+    {
+        $this->shouldImplement(OrderItemPromotionAdjustmentsApplicatorInterface::class);
+    }
+
+    function it_applies_promotion_adjustments_on_all_units_of_given_order_item(
+        OrderItemInterface $orderItem,
+        OrderItemUnitInterface $unit1,
+        OrderItemUnitInterface $unit2,
+        PromotionInterface $promotion,
+        AdjustmentFactoryInterface $adjustmentFactory,
+        AdjustmentInterface $firstAdjustment,
+        AdjustmentInterface $secondAdjustment
+    ) {
+        $promoName = 'Winter guns promotion!';
+        $promoCode = 'WINTER_GUNS_PROMOTION';
+        $orderItem
+            ->getUnits()
+            ->willReturn(new ArrayCollection([$unit1->getWrappedObject(), $unit2->getWrappedObject()]))
+        ;
+
+        $promotion->getName()->willReturn($promoName);
+        $promotion->getCode()->willReturn($promoCode);
+
+        $adjustmentFactory
+            ->createNew()
+            ->willReturn($firstAdjustment, $secondAdjustment)
+        ;
+
+        $firstAdjustment->setType(AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT)->shouldBeCalled();
+        $firstAdjustment->setLabel($promoName)->shouldBeCalled();
+        $firstAdjustment->setOriginCode($promoCode)->shouldBeCalled();
+        $firstAdjustment->setAmount(-1000)->shouldBeCalled();
+
+        $secondAdjustment->setType(AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT)->shouldBeCalled();
+        $secondAdjustment->setLabel($promoName)->shouldBeCalled();
+        $secondAdjustment->setOriginCode($promoCode)->shouldBeCalled();
+        $secondAdjustment->setAmount(-1000)->shouldBeCalled();
+
+        $unit1->addAdjustment($firstAdjustment)->shouldBeCalled();
+        $unit1->getTotal()->shouldBeCalled()->willReturn(5000);
+        $unit2->addAdjustment($secondAdjustment)->shouldBeCalled();
+        $unit2->getTotal()->shouldBeCalled()->willReturn(5000);
```

Comment:
There is no need to call `shouldBeCalled()` and `willReturn(...)` simultaneously. `willReturn(...)` is enough. Here and everywhere else.